framework/include/problems/FEProblemBase.h

```diff
@@ -1610,6 +1610,11 @@ class FEProblemBase : public SubProblem, public Restartable
    */
   bool useSNESMFReuseBase() { return _snesmf_reuse_base; }
 
+  /**
+   * Return a flag that indicates if we want to skip exception and stop solve
+   */
+  bool & skipExceptionCheck() { return _skip_exception_check; }
```

Comment:
Since you aren't reading this value externally (there's no need to). Let's just make this a "setter" only. I would prefer to see it as returning `void`. Then we don't have to track down somebody running around with a writable reference later. Keep our API as closed as possible.